tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/PostResourceIntTest.java

```diff
@@ -397,6 +399,164 @@ public void shouldAllowMMultipleOtherSpecialties() throws Exception {
+    // the Post needs a Primary specialty as well, otherwise it wouldn't be possible to set a
+    // sub_specialty
+    post = createEntity();
+    PostSpecialty primaryPostSpecialty = createPostSpecialty(specialty, PostSpecialtyType.PRIMARY, post);
+    post.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty)));
+    postRepository.saveAndFlush(post);
+
+    // Attempt to update a Post that has a specialty of specialtyType PLACEMENT as a subspecialty
+    // (i.e. link Post and a Specialty of SpecialtyType.PLACEMENT via a PostSpecialty that is
+    // of PostSpecialtyType.SUB_SPECIALTY)
+    // The update should fail.
+
+    Post updatedPost = createEntity();
+    updatedPost.setId(post.getId());
+
+    Specialty notASubspecialty = createSpecialty();
+    notASubspecialty.setSpecialtyTypes(new HashSet<>(
+        Collections.singletonList(SpecialtyType.PLACEMENT)));
+    em.persist(notASubspecialty);
+
+    PostSpecialty subspecialtyPostSpecialty = createPostSpecialty(notASubspecialty,
+        PostSpecialtyType.SUB_SPECIALTY, post);
+    post.getSpecialties().stream().findFirst().ifPresent(ps -> primaryPostSpecialty.setId(ps.getId()));
+    updatedPost.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty, subspecialtyPostSpecialty)));
+
+    PostDTO updatedPostDto = postMapper.postToPostDTO(updatedPost);
+
+    restPostMockMvc.perform(put("/api/posts")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(TestUtil.convertObjectToJsonBytes(updatedPostDto)))
+        .andExpect(status().isBadRequest())
+        .andExpect(jsonPath("$.message").value("error.validation"))
+        .andExpect(jsonPath("$.fieldErrors[0].field").value("specialties"));
+
+    Post postAfterUpdate = postRepository.findById(post.getId()).orElse(null);
+    Set<PostSpecialty> updatedPostSpecialtySet = postAfterUpdate.getSpecialties();
+    List <PostSpecialty> addedSubspecialties = updatedPostSpecialtySet.stream()
+        .filter(ps -> ps.getPostSpecialtyType().equals(PostSpecialtyType.SUB_SPECIALTY))
+        .collect(Collectors.toList());
+
+    assertThat(addedSubspecialties.size()).isZero();
+    assertThat(postAfterUpdate).isEqualTo(post);
+  }
+
+  @Test
+  @Transactional
+  public void shouldAllowUpdateOfPostIfSubspecialtyIsNotASubspecialty() throws Exception {
+    // initialize database with a Post
+    // the Post needs a Primary specialty as well, otherwise it wouldn't be possible to set a
+    // sub_specialty
+    post = createEntity();
+    PostSpecialty primaryPostSpecialty = createPostSpecialty(specialty, PostSpecialtyType.PRIMARY, post);
+    post.setSpecialties(new HashSet<>(Arrays.asList(primaryPostSpecialty)));
+    postRepository.saveAndFlush(post);
+
+    // Attempt to update a Post that has a specialty of specialtyType SUB_SPECIALTY as a subspecialty
+    // (i.e. link Post and a Specialty that is SpecialtyType.SUB_SPECIALTY in a PostSpecialty in
+    // a PostSpecialty of PostSpecialtyType.SUB_SPECIALTY)
```

Comment:
Looks like a bit of word duplication 'in a PostSpecialty in a PostSpecialty' which makes this seem possibly a bit confusing.

Reply:
Ah yes, I copied this more than once, then went back to edit them, and in the end they got all messed up.
I've rewritten all four tests' comments, hopefully they're clearer now.